src/index/IndexMetaData.cpp

```diff
@@ -58,42 +64,143 @@ void IndexMetaData::createFromByteBuffer(unsigned char* buf) {
 }
 
 // _____________________________________________________________________________
-const RelationMetaData IndexMetaData::getRmd(Id relId) const {
-  auto it = _data.find(relId);
-  AD_CHECK(it != _data.end());
-  RelationMetaData ret(it->second);
-  if (it->second.hasBlocks()) {
-    ret._rmdBlocks = &_blockData.find(it->first)->second;
+// specialization for MMapBased Arrays which only read
+// block-based data from Memory
+template <>
+void IndexMetaDataMmap::createFromByteBuffer(unsigned char* buf) {
+  size_t nameLength = *reinterpret_cast<size_t*>(buf);
+  size_t nofBytesDone = sizeof(size_t);
+  _name.assign(reinterpret_cast<char*>(buf + nofBytesDone), nameLength);
+  nofBytesDone += nameLength;
+  // skip nOfRelations, since this information is already stored in
```

Comment:
because we first have to run the IndexBuilder to create MetaData which is then written to disk which we then read again when opening the server using createFromByteBuffer.
The local nofRelations variable is needed in the  non-persistent specializations of IndexMetaData where we read the FullRelationalMetaData from our byteBuffer. (You can check  the difference  to  the "standard" implementation.
I also added a (hopefully) better comment to this.